### ISSUE TITLE
Alias usages not detected from keywords in quoted forms

### ISSUE THREAD

**Issue by milankinen** [borkdude/clj-kondo#981]
Hi and thanks for this great tool! Here is a small issue we found while using it:

**version**

clj-kondo v2020.07.29

**platform**

Native version / OSX and Linux

**problem**

It seems that alias usages from quoted forms are not registered. If alias is used only in quoted form (real life example case: Datomic queries), clj-kondo gives `warning: namespace <ns> is required but never used`. However, if we remove the require, the code won't work anymore because reader can't recognize keyword without it. 

We've solve this by so far by using fully qualified keywords but it's a bit annoying because some of the namespaces can be quite long. 

**repro**

Here is a very basic code reproducing the issue (uses default configs):

```clj
(ns org.tsers.example
  (:require [datomic.api :as d]
            [org.tsers.user :as user]))

(defn find-users-by-country [db country]
  (d/q '[:find [?user-id ...]
         :in $ ?country 
         :where [?user ::user/country ?country]
                [?user ::user/id ?user-id]]
        db country))
```

**expected behavior**

I'd expect the code to be linted without warnings, instead I get 

```
$ ./clj-kondo --lint example.clj 
example.clj:3:14: warning: namespace org.tsers.user is required but never used
linting took 10ms, errors: 0, warnings: 1
```

**Comment by borkdude** [comment]
A small workaround for now is to add a top-level keyword `::user/foo` or use this config:

https://github.com/borkdude/clj-kondo/blob/master/doc/config.md#exclude-required-but-unused-namespace-from-being-reported

**Comment by milankinen** [comment]
Yes we've indeed considered those as well but decided not to use them because of the possibility to accidentally cause false negatives by forgetting the references/exclusions while doing code refactors. Having to write the fully qualified keyword name is a bit annoying but at least it's robust over time and easier to maintain.

**Comment by borkdude** [comment]
Fixed on master.

**Comment by milankinen** [comment]
Wow that was fast! Thank you!
